draft-ietf-quic-multipath.md

```diff
@@ -790,32 +790,71 @@ In this example the client detects the network environment change
-Afterwards the connection IDs in both directions can be retired
-using the RETIRE_CONNECTION_ID frame.
+{{fig-example-path-close1}} illustrates an example of path closing when both
+the client and the server use non-zero-length CIDs. For the first path, the
+server's 1-RTT packets use DCID C1, which has a sequence number of 1; the
+client's 1-RTT packets use DCID S2, which has a sequence number of 2. For the
+second path, the server's 1-RTT packets use DCID C2, which has a sequence
+number of 2; the client's 1-RTT packets use CID S3, which has a sequence number
+of 3. Note that two paths use different packet number spaces. In this case, the
+client is going to close the first path. It identifies the path by the sequence
+number of the received packet's DCID over that path (path identifier type
+0x00), hence using a path_id 1. In reply, the server confirms the path using
+the sequence number of the received packet's DCID over that path (path
+identifier type 0x00), which corresponds to a path_id 2. Both the client and
+the server can close the path after receiving the RETIRE_CONNECTION_ID frame
+for that path.
 
 ~~~
 Client                                                      Server
 
 (client tells server to abandon a path)
-1-RTT[X]: DCID=S2 PATH_ABANDON[path_id=1]->
+1-RTT[X]: DCID=S2 PATH_ABANDON[path_id_type=0, path_id=1]->
                            (server tells client to abandon a path)
-  <-1-RTT[Y]: DCID=C1 PATH_ABANDON[path_id=2], ACK_MP[Seq=2, PN=X]
-(client abandons the path that it is using)
+      <-1-RTT[Y]: DCID=C1 PATH_ABANDON[path_id_type=0, path_id=2],
+                                               ACK_MP[Seq=2, PN=X]
+(client retires the corresponding CID)
 1-RTT[U]: DCID=S3 RETIRE_CONNECTION_ID[2], ACK_MP[Seq=1, PN=Y] ->
-                       (server abandons the path that it is using)
+                            (server retires the corresponding CID)
  <- 1-RTT[V]: DCID=C2 RETIRE_CONNECTION_ID[1], ACK_MP[Seq=3, PN=U]
 ~~~
-{: #fig-example-path-close title="Example of closing a path (path id type=0x00)"}
+{: #fig-example-path-close1 title="Example of closing a path when both the
+client and the server choose to receive non-zero-length CIDs."}
+
+{{fig-example-path-close2}} illustrates an example of path closing when the
+client chooses to receive zero-length CIDs while the server chooses to receive
+non-zero-length CIDs.  Because there is a zero-length CID in one direction,
+single packet number spaces are used. For the first path, the client's 1-RTT
+packets use DCID S2, which has a sequence number of 2. For the second path, the
+client's 1-RTT packets use DCID S3, which has a sequence number of 3. Again, in
+this case, the client is going to close the first path. Because the client now
+receives zero-length CID packets, it needs to use path identifier type 0x01,
+which identifies a path by the DCID sequence number of the packets it sends
+over that path, and hence, it uses a path_id 2 in its PATH_ABANDON frame. The
```

Comment:
Similarly, Section 11.1 currently states

```
Type 1: Refer to the connection identifier used by the receiver of the control frame
when sending data over the specified path. This method MUST NOT be used if this
connection identifier is zero-length.
```

As I understand this, the example should use type 0, no?